src/handler/http/request/dashboards/mod.rs

```diff
@@ -249,26 +293,60 @@ async fn delete_dashboard(path: web::Path<(String, String)>, req: HttpRequest) -
 async fn move_dashboard(
     path: web::Path<(String, String)>,
     folder: web::Json<MoveDashboard>,
-) -> Result<HttpResponse, Error> {
+) -> impl Responder {
     let (org_id, dashboard_id) = path.into_inner();
-    if folder.from.is_empty() || folder.to.is_empty() {
-        return Ok(
-            HttpResponse::InternalServerError().json(MetaHttpResponse::message(
-                http::StatusCode::BAD_REQUEST.into(),
-                "Please specify from & to folder from dashboard movement".to_string(),
-            )),
-        );
-    };
-
-    let resp =
-        match dashboards::move_dashboard(&org_id, &dashboard_id, &folder.from, &folder.to).await {
-            Ok(resp) => resp,
-            Err(_) => HttpResponse::InternalServerError().into(),
-        };
-    Ok(resp)
+    match dashboards::move_dashboard(&org_id, &dashboard_id, &folder.from, &folder.to).await {
+        Ok(()) => HttpResponse::Ok().json(MetaHttpResponse::message(
+            http::StatusCode::OK.into(),
+            "Dashboard moved".to_string(),
+        )),
+        Err(err) => err.into(),
+    }
 }
 
 fn get_folder(req: HttpRequest) -> String {
     let query = web::Query::<HashMap<String, String>>::from_query(req.query_string()).unwrap();
     crate::common::utils::http::get_folder(&query)
 }
+
+/// Parses the bytes into a dashboard with the given version.
+fn parse_dashboard(bytes: web::Bytes) -> Result<Dashboard, serde_json::Error> {
```

Comment:
Previously this logic was in the Service Layer since previously we were passing unparsed bytes from this module into the Service Layer.

Since parsing the bytes of an HTTP request body is strictly the domain of the HTTP layer, this logic has been moved here. Now the HTTP Layer parses the request body bytes into the `Dashboard` domain model and then passes that model on to the Service Layer.